docker-images/operator/scripts/launch_java.sh

```diff
@@ -12,6 +14,70 @@ function get_gc_opts {
   fi
 }
 
+# This functions calculates the max memory that the container could using cgroups V2
+max_memory_using_cgroups_v2() {
+  # High number which is the max limit until which memory is supposed to be
+  # unbounded.
+  local max_mem_cgroup=0
+  local max_mem_meminfo_kb
+  max_mem_meminfo_kb="$(< /proc/meminfo awk '/MemTotal/ {print $2}')"
+  local max_mem_meminfo
+  max_mem_meminfo=$((max_mem_meminfo_kb*1024))
+
+   max_mem_cgroup="$(cat ${mem_file_cgroups_v2})"
+
+  if [ "${max_mem_cgroup:-0}" != -1 ] && [ "${max_mem_cgroup:-0}" -lt ${max_mem_meminfo:-0} ]; then
+    echo "${max_mem_cgroup}"
+  fi
+}
+
+# set an enviroment variable CONTAINER_MAX_MEMORY with the memory limit of the container
+init_memory_limit_env_vars() {
+  # Read in container limits and export the as environment variables
+  local mem_limit
+  mem_limit="$(max_memory_using_cgroups_v2)"
+  if [ -n "${mem_limit}" ]; then
+    export CONTAINER_MAX_MEMORY="${mem_limit}"
+  fi
+}
+
+calc_mem_opt() {
+  local max_mem="$1"
+  local fraction="$2"
+  local mem_opt="$3"
+
+  local val
+  val=$(calc "round($1*$2/100/1048576)" "${max_mem}" "${fraction}")
+  echo "-X${mem_opt}${val}m"
+}
+
+# Generic formula evaluation based on awk
+calc() {
+  local formula="$1"
+  shift
+  echo "$@" | awk '
+    function round(x) {
+      return int(x + 0.5)
+    }
+    {print '"int(${formula})"'}
+  '
+}
+
+# Calculate the value of -Xmx options base on CONTAINER_MAX_MEMORY
+calc_max_memory() {
+  if [ -z "${CONTAINER_MAX_MEMORY:-}" ]; then
+    return
+  fi
+
+  if [ "${CONTAINER_MAX_MEMORY}" -le 314572800 ]; then
+    # Restore the one-fourth default heap size instead of the one-half below 300MB threshold
+    # See https://docs.oracle.com/javase/8/docs/technotes/guides/vm/gctuning/parallel.html#default_heap_size
+    calc_mem_opt "${CONTAINER_MAX_MEMORY}" "50" "mx"
+  else
+    calc_mem_opt "${CONTAINER_MAX_MEMORY}" "75" "mx"
+  fi
```

Comment:
Why these numbers? There is a reason why we set the MaxRamPercentage to 20% => because it does not work well with higher numbers. So shouldn't this follow the same logic?

Reply:
Taking a look at the deployment that used this code,

https://github.com/strimzi/strimzi-kafka-operator/blob/main/helm-charts/helm3/strimzi-kafka-operator/templates/060-Deployment-strimzi-cluster-operator.yaml#L50

There is only one container in the pod deployed by as part of that deployment. So we don't see the need to use only 20% of the memory available by the container for that java application (operator).

Reply:
The 20% value is based on our experience (likely with cgroups v1). Why do you think with cgroups v2 the Java requirements will suddenly change and it will be able to use 50% or 75%?

Reply:
More likely I'm missing something, but as I see it, It's not a matter of using cgroups v1 or v2, it's a matter of the point of view of the application inside k8s.

The deployment of the operator only has 1 container inside the pod created by k8s, strimzi-cluster-operator, which is a Java Application, so the question could be:
Why only give that application 20% of the memory requested to k8s? There isn't anything else that will be executed in that pod.

So perhaps the value that should be re-evaluated is the default request to k8s. 
This one:
https://github.com/strimzi/strimzi-kafka-operator/blob/main/helm-charts/helm3/strimzi-kafka-operator/values.yaml#L142

Reply:
Thanks for the explanation. For us, those values are working fine.

But since you have more information about the behaviour of the operator, we are going to keep this value aligned with MaxRAMPercentage.

calc_mem_opt "${CONTAINER_MAX_MEMORY}" "20"

Also, bear in mind that this workaround must be removed once the java version used by the operator is updated with a version without that issue.